docs/src/main/asciidoc/deploying-to-kubernetes.adoc

```diff
@@ -860,14 +860,63 @@ implementation("io.quarkus:quarkus-kubernetes-client")
 ----
 
 To access the API server from within a Kubernetes cluster, some RBAC related resources are required (e.g. a ServiceAccount, a RoleBinding).
-So, when the `kubernetes-client` extension is present, the `kubernetes` extension is going to create those resources automatically, so that application will be granted the `view` role.
-If more roles are required, they will have to be added manually.
+So, when the `kubernetes-client` extension is present, by default the `kubernetes` extension is going to create those resources automatically, so that application will be granted the `view` role. You can fully customize the roles and subjects to use using the properties under `quarkus.kubernetes-client.role-binding`.
+If more roles are required, they can create them using the properties under `quarkus.kubernetes.rbac.roles`.
 
 [NOTE]
 ====
 You can disable the RBAC resources generation using the property `quarkus.kubernetes-client.generate-rbac=false`.
 ====
 
+=== Generating RBAC resources
+
+In some scenarios, it's necessary to generate additional https://kubernetes.io/docs/reference/access-authn-authz/rbac/[RBAC] resources. The Kubernetes extension got you covered with the `quarkus.kubernetes.rbac` properties:
+
+[source,properties]
+----
+# Generate Role resource with name "my-role" <1>
+quarkus.kubernetes.rbac.roles.my-role.policy-rules.0.api-groups=extensions,apps
+quarkus.kubernetes.rbac.roles.my-role.policy-rules.0.resources=deployments,deployments/status
+quarkus.kubernetes.rbac.roles.my-role.policy-rules.0.verbs=get,watch,list
+quarkus.kubernetes.rbac.roles.my-role.policy-rules.0.resources=services
+
+# Generate ServiceAccount resource with name "my-service-account" in namespace "my_namespace" <2>
+quarkus.kubernetes.rbac.service-accounts.my-service-account.namespace=my_namespace
+
+# Bind Role "my-role" with ServiceAccount "my-service-account"
+quarkus.kubernetes.rbac.role-bindings.my-role-binding.subjects.my-service-account.kind=ServiceAccount
+quarkus.kubernetes.rbac.role-bindings.my-role-binding.subjects.my-service-account.namespace=my_namespace
+quarkus.kubernetes.rbac.role-bindings.my-role-binding.role-name=my-role
+quarkus.kubernetes.rbac.role-bindings.my-role-binding.cluster-wide=false
+
+# Use the service account "my-service-account" in the application
+quarkus.kubernetes.service-account=my-service-account <3>
```

Comment:
I am wondering if this could be inferred.

We could possibly generated namespace when a namespace has been defined for the deployment or is referred by the rules.

The only challenge is what happens, if we want to refer to an externally created `ServiceAccount`. However, this is not our job and on top of that what we generate should be standalone.

TLDR: This should be optional.

Reply:
I would actually avoid the "inferred" configuration, so it should be up to the user to use one service account or another.

Reply:
We should always use the application service account, unless explicitly specified.
There is absolutely no reason specifying 2 full lines of configuration for something that should be the application service account for most of the cases.